pdfgen/src/lib.rs

```diff
@@ -35,20 +43,38 @@ impl Default for Document {
         Self {
             catalog,
             id_manager,
+            pages: Vec::new(),
         }
     }
 }
 
 impl Document {
+    /// Creates a new page inside the document.
+    pub fn create_page(&mut self, media_box: impl Into<Rectangle>) -> &mut Page {
+        let id = self.id_manager.create_id();
+        self.catalog.page_tree_mut().add_page(id.clone());
+        self.pages.push(Page::new(
+            id,
+            self.catalog.page_tree().obj_ref(),
+            media_box.into(),
+        ));
+
+        self.pages.last_mut().unwrap()
+    }
+
     /// Write the PDF contents into the provided writer.
     pub fn write(&self, writer: &mut impl Write) -> Result<(), io::Error> {
         let mut pdf_writer = PdfWriter::new(writer);
         pdf_writer.write_header()?;
 
         pdf_writer.write_object(&self.catalog, self.catalog.obj_ref())?;
         pdf_writer.write_object(self.catalog.page_tree(), self.catalog.page_tree().obj_ref())?;
+        for page in &self.pages {
```

Comment:
Nitpick about spacing 🫣 

```suggestion

        for page in &self.pages {
```